src/main/scala/viper/gobra/frontend/info/implementation/property/Comparability.scala

```diff
@@ -25,6 +25,13 @@ trait Comparability extends BaseProperty { this: TypeInfoImpl =>
     case _ => false
   }
 
+  lazy val ghostComparableTypes: Property[(Type, Type)] = createFlatProperty[(Type, Type)] {
+    case (left, right) => s"$left is not comparable in ghost with $right"
+  } {
+    case (Single(left), Single(right)) => assignableTo(left, right) || assignableTo(right, left)
```

Comment:
I was initially confused that not more checks are neccessary, but I guess the `(NilType, r) if isPointerType(r)` check is not necessary since this usually makes comparisons of uncomparable types with nil possible.